hibernate-core/src/test/java/org/hibernate/engine/internal/StatisticalLoggingSessionEventListenerTest.java

```diff
@@ -0,0 +1,226 @@
+		assertThat( executingJdbcBatchesMetric.getCount() ).isEqualTo( 1 );
+
+		// performing L2C puts
+		SessionMetric performingL2CPutsMetric = extractMetric(
+				sessionMetricsLog,
+				"([0-9]+) nanoseconds spent performing ([0-9]+) L2C puts"
+		);
+		assertThat( performingL2CPutsMetric.getDuration() ).isGreaterThan( 0L )
+				.isLessThan( sessionNanoDuration );
+		assertThat( performingL2CPutsMetric.getCount() ).isEqualTo( 4 );
+
+		// performing L2C hits
+		SessionMetric performingL2CHitsMetric = extractMetric(
+				sessionMetricsLog,
+				"([0-9]+) nanoseconds spent performing ([0-9]+) L2C hits"
+		);
+		assertThat( performingL2CHitsMetric.getDuration() ).isGreaterThan( 0L )
+				.isLessThan( sessionNanoDuration );
+		assertThat( performingL2CHitsMetric.getCount() ).isEqualTo( 2 );
+
+		// performing L2C misses
+		SessionMetric performingL2CMissesMetric = extractMetric(
+				sessionMetricsLog,
+				"([0-9]+) nanoseconds spent performing ([0-9]+) L2C misses"
+		);
+		assertThat( performingL2CMissesMetric.getDuration() ).isGreaterThan( 0L )
+				.isLessThan( sessionNanoDuration );
+		assertThat( performingL2CMissesMetric.getCount() ).isEqualTo( 1 );
+
+		// executing flushes
+		SessionMetric executingFlushesMetric = extractMetric(
+				sessionMetricsLog,
+				"([0-9]+) nanoseconds spent executing ([0-9]+) flushes"
+		);
+		assertThat( executingFlushesMetric.getDuration() ).isGreaterThan( 0L )
+				.isLessThan( sessionNanoDuration );
+		assertThat( executingFlushesMetric.getCount() ).isEqualTo( 1 );
+
+		// executing pre-partial-flushes
+		SessionMetric executingPrePartialFlushesMetric = extractMetric(
+				sessionMetricsLog,
+				"([0-9]+) nanoseconds spent executing ([0-9]+) pre-partial-flushes"
+		);
+		assertThat( executingPrePartialFlushesMetric.getDuration() ).isGreaterThan( 0L )
+				.isLessThan( sessionNanoDuration );
+		assertThat( executingPrePartialFlushesMetric.getCount() ).isEqualTo( 2 );
+
+		// executing partial-flushes
+		SessionMetric executingPartialFlushesMetric = extractMetric(
+				sessionMetricsLog,
+				"([0-9]+) nanoseconds spent executing ([0-9]+) partial-flushes"
+		);
+		assertThat( executingPartialFlushesMetric.getDuration() ).isGreaterThan( 0L )
+				.isLessThan( sessionNanoDuration );
+		assertThat( executingPartialFlushesMetric.getCount() ).isEqualTo( 2 );
+
+		// Number of metrics
+		final int numberOfMetrics = 11;
+		List<SessionMetric> metricList = List.of(
+				acquiringJdbcConnectionsMetric,
+				releasingJdbcConnectionsMetric,
+				preparingJdbcStatmentsMetric,
+				executingJdbcStatmentsMetric,
+				executingJdbcBatchesMetric,
+				performingL2CPutsMetric,
+				performingL2CHitsMetric,
+				performingL2CMissesMetric,
+				executingFlushesMetric,
+				executingPrePartialFlushesMetric,
+				executingPartialFlushesMetric
+		);
+		assertThat( metricList.size() ).isEqualTo( numberOfMetrics );
+		// Number of lines
+		assertThat( sessionMetricsLog.lines().count() ).isEqualTo( numberOfMetrics + 2 );
```

Comment:
This is done in order to make the test fail if new metrics are added to log event.

Reply:
I would remove this check or add a message that makes easy to understand what is causing the failure

Reply:
I improved the failure message. It should be clear now that we check for the number of lines in the log to be equal to the number of metrics plus header and footer line